api/handlers/search_handler.go

```diff
@@ -74,71 +72,40 @@ func (handler *SearchHandler) buildSearchCfg(params url.Values) (cfg models.Sear
 	cfg.Text = text
 	cfg.MaxResults, _ = strconv.Atoi(params.Get("size"))
 	cfg.Filters = filterConfigFromValues(params)
-	cfg.TypeWhiteList = parseTypeWhiteList(params)
+	cfg.TypeWhiteList, err = parseTypeWhiteList(params)
+	if err != nil {
+		return
+	}
+
 	return
 }
 
-func (handler *SearchHandler) toSearchResponse(ctx context.Context, results []models.SearchResult) (response []SearchResponse, err error) {
-	typeRepo := newCachingTypeRepo(handler.typeRepo)
-	for _, result := range results {
-		recordType, err := typeRepo.GetByName(ctx, result.TypeName)
-		if err != nil {
-			return nil, fmt.Errorf("typeRepository.GetByName: %q: %v", result.TypeName, err)
-		}
-
+func (handler *SearchHandler) toSearchResponse(ctx context.Context, records []record.Record) (response []SearchResponse, err error) {
+	for _, r := range records {
 		res := SearchResponse{
-			Type:        recordType.Name,
-			ID:          result.Record.Urn,
-			Title:       result.Record.Name,
-			Description: result.Record.Description,
-			Labels:      result.Record.Labels,
-			Service:     result.Record.Service,
+			ID:          r.Urn,
+			Title:       r.Name,
+			Description: r.Description,
+			Labels:      r.Labels,
+			Service:     r.Service,
+			Type:        string(r.Type),
```

Comment:
I saw we have `String` function for `Type`. Why not using that instead?
```suggestion
			Type:        r.Type.String(),
```

Reply:
yeah suree, `String()` was added late and forgot some codes were still using `string(type)`.
updating 👌

Reply:
updated https://github.com/odpf/columbus/pull/25/commits/c3e8a36394323d03afd6cee665592248e92a4ebf